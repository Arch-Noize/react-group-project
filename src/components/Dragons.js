import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addReservation, cancelReservation } from '../redux/dragons/dragonSlice';
import styles from '../styles/Dragons.module.css';

const Dragons = () => {
  const dragons = useSelector((state) => state.dragons.dragons);
  const dispatch = useDispatch();

  return (
    <>
      <ul className={styles.dragonCard}>
        {dragons.map((dragon) => (
          <li key={dragon.id} className={styles.dragonList}>
            <img src={dragon.img} alt="" className={styles.image} />
            <h2>{dragon.name}</h2>
            <p>{dragon.desc}</p>
            {dragon.reserved ? (
              <button type="button" onClick={() => dispatch(cancelReservation(dragon.id))}>
                Cancel Dragon
              </button>
            ) : (
              <button type="button" onClick={() => dispatch(addReservation(dragon.id))}>
                Reserve Dragon
              </button>
            )}
          </li>
        ))}
      </ul>
    </>
  );
};

export default Dragons;
