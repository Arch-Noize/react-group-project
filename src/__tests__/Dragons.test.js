import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Dragons from '../components/Dragons';

const mockStore = configureMockStore([]);

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('Loading dragons component', () => {
  it('renders correctly', () => {
    const initialState = {
      dragons: {
        dragons: [
          {
            id: 1,
            name: 'Dragon 1',
            type: 'Type 1',
            flickr_images: ['image1.jpg'],
            reserved: false,
          },
        ],
      },
    };

    const store = mockStore(initialState);

    const { tree } = render(
      <Provider store={store}>
        <Dragons />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
