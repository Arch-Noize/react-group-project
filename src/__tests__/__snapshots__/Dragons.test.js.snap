// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading dragons component renders correctly 1`] = `undefined`;
